src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -606,6 +642,7 @@ def load_status_file_components(self, initial_load=False):
                         self.__installation_total_error_count = self.__get_total_error_count_from_prev_status(errors['message'])
             if name == Constants.PATCH_ASSESSMENT_SUMMARY:     # if it exists, it must be to spec, or an exception will get thrown
                 self.__assessment_summary_json = self.__get_substatus_message(complete_status_file_data, i)
+                # Reload patches into assessment ordered map for fast look up
```

Comment:
As Koshy suggested in one of his reviews, please avoid adding changes that are not about the functionality being implemented in a PR. I was trying to discount the ones I have seen in this PR so far, thinking they were just a few, but it seems there's quite some . Please restrict your PR changes to only what the PR is trying to achieve, all of the other good to have changes can come in, in one separate PR.

Reply:
its' counterproductive to revert this change, this logic is very small and straightforward, so I refactor the logic into a function and reuse it in truncation to avoid replication.

Reply:
Ignore this for now to look at the bigger picture. Rajasi's guidance generally applies but I'm resolving to reduce noise.